cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/constraint/TimeRangeConstraintTest.java

```diff
@@ -45,54 +45,102 @@
   private static final WorkflowId WORKFLOW_ID = APP_ID.workflow("wf1");
   private static final DatasetId DATASET_ID = TEST_NS.dataset("pfs1");
 
+  private static final ProgramSchedule SCHEDULE = new ProgramSchedule("SCHED1", "one partition schedule", WORKFLOW_ID,
+                                                                      ImmutableMap.of("prop3", "abc"),
+                                                                      new PartitionTrigger(DATASET_ID, 1),
+                                                                      ImmutableList.<Constraint>of());
+
   @Test
   public void testInit() {
     TimeRangeConstraint timeRangeConstraint = new TimeRangeConstraint("16:00", "17:00", TimeZone.getTimeZone("PST"));
     Assert.assertEquals("PST", timeRangeConstraint.getTimeZone());
-  }
 
-  @Test
-  public void test() {
     // simple construction should work
     new TimeRangeConstraint("03:16", "03:17", TimeZone.getDefault());
+    // leading zero shouldn't be necessary
+    new TimeRangeConstraint("03:16", "3:17", TimeZone.getDefault());
```

Comment:
I mean, add this line:

new TimeRangeConstraint("03:17", "03:16", TimeZone.getDefault());

to validate that we accept a range where start > end